src/GovUk.Education.ExploreEducationStatistics.Content.Model/MethodologyParent.cs

```diff
@@ -10,18 +12,56 @@ public class MethodologyParent
 
         public List<PublicationMethodology> Publications { get; set; }
 
+        // TODO SOW4 EES-2375 Move slug to parent
+        // Need to sort migration for this field
+        [NotMapped]
+        public string Slug { get; set; }
+
         public List<Methodology> Versions { get; set; }
 
         public Methodology LatestVersion()
         {
-            return Versions?.SingleOrDefault(mv => IsLatestVersionOfMethodology(mv.Id));
+            if (Versions == null)
+            {
+                throw new ArgumentException("Methodology must be hydrated with Versions to get the latest version");
+            }
+
+            return Versions.SingleOrDefault(mv => IsLatestVersionOfMethodology(mv.Id));
+        }
+
+        public Methodology LatestPublishedVersion()
+        {
+            if (Versions == null)
+            {
+                throw new ArgumentException("Methodology must be hydrated with Versions to get the latest published version");
+            }
+
+            return Versions.SingleOrDefault(mv => IsLatestPublishedVersionOfMethodology(mv.Id));
         }
 
         public bool IsLatestVersionOfMethodology(Guid methodologyVersionId)
         {
-            return Versions != null
-                   && Versions.Exists(mv => mv.Id == methodologyVersionId)
+            if (Versions == null)
+            {
+                throw new ArgumentException("Methodology must be hydrated with Versions to test the latest version");
+            }
+
+            return Versions.Exists(mv => mv.Id == methodologyVersionId)
                    && Versions.All(mv => mv.PreviousVersionId != methodologyVersionId);
         }
+
+        public bool IsLatestPublishedVersionOfMethodology(Guid methodologyVersionId)
+        {
+            if (Versions == null)
+            {
+                throw new ArgumentException("Methodology must be hydrated with Versions to test the latest published version");
+            }
+
+            var version = Versions.SingleOrDefault(mv => mv.Id == methodologyVersionId);
+
+            return version is {PubliclyAccessible: true}
```

Comment:
Ooh what's this syntax?   `version is {PubliclyAccessible: true}`

Is there a reason why not to do `version.PubliclyAccessible` instead?

Reply:
We do want to do `version.PubliclyAccessible` but we also want to check for version being null too as the `id` passed to this method could theoretically be none of the versions and completely unrelated so we should return false if that's the case.

So this `is` syntax with the pattern expression is null-safe and is an alternative of 

`version != null && version.PubliclyAccessible`

I didn't know about it but the IDE suggested it as soon as I wrote it so I accepted it 🙂.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/is

Reply:
Nice one!  I guess is this also the equivalent of version?.PubliclyAccessible   ?